modules/appium/src/main/java/com/codeborne/selenide/appium/SelenideAppiumPageFactory.java

```diff
@@ -44,15 +52,25 @@ public class SelenideAppiumPageFactory extends SelenidePageFactory {
   @Nonnull
   @CheckReturnValue
   protected By findSelector(Driver driver, Field field) {
-    AppiumByBuilder builder = byBuilder(driver);
+    AppiumByBuilder builder = byBuilder(driver, field);
     builder.setAnnotated(field);
     By selector = builder.buildBy();
     return selector != null ? selector : super.findSelector(driver, field);
   }
 
+
   @Nonnull
   @CheckReturnValue
-  private DefaultElementByBuilder byBuilder(Driver driver) {
+  private DefaultElementByBuilder byBuilder(Driver driver, Field field) {
+    if (!isPlatformAnnotationAdded(field)) {
+      return new DefaultElementByBuilder(null, null);
+    }
+
+    if (!driver.hasWebDriverStarted()) {
+      throw new RuntimeException("The Appium Page factory requires a browser instance to be created before calling" +
```

Comment:
I don't think we really need it. Either way, users will get  `No webdriver is bound to current thread: 1. You need to call open(url) first.`
And wording `The Appium Page factory requires a browser instance` may confuse. Which browser do we ask to launch when we test a mobile application?

Reply:
I thought it would be helpful to handle such cases better by adding additional context to aid in debugging. Initially, it wasn’t clear why I received the exception, and I wasn’t sure if it was expected.

I've updated the exception message. Please review it, and if my changes don't align with your perspective, let me know, and I’ll remove this part.